src/WiFiCredentials.cpp

```diff
@@ -0,0 +1,194 @@
+#include "WiFiCredentials.h"
+
+#include <Arduino.h>
+#include <LittleFS.h>
+
+#include <esp_log.h>
+
+#include <cstring>
+
+const char* const TAG = "WiFiCredentials";
+
+using namespace OpenShock;
+
+bool WiFiCredentials::Load(std::vector<WiFiCredentials>& credentials) {
+  credentials.clear();
+
+  // Ensure the /wifi/creds directory exists
+  if (!LittleFS.exists("/wifi/creds")) {
+    if (!LittleFS.exists("/wifi")) {
+      if (!LittleFS.mkdir("/wifi")) {
+        ESP_LOGE(TAG, "Failed to create /wifi directory");
+        return false;
+      }
+    }
+    if (!LittleFS.mkdir("/wifi/creds")) {
+      ESP_LOGE(TAG, "Failed to create /wifi/creds directory");
+      return false;
+    }
+    ESP_LOGI(TAG, "No credentials directory found, created one");
+    return true;
+  }
+
+  File credsDir = LittleFS.open("/wifi/creds");
+  if (!credsDir) {
+    ESP_LOGE(TAG, "Failed to open /wifi/creds");
+    return false;
+  }
+
+  while (true) {
+    File file = credsDir.openNextFile();
+    if (!file) {
+      break;
+    }
+
+    WiFiCredentials creds;
+    if (!creds._load(file)) {
+      ESP_LOGE(TAG, "Failed to load credentials from %s", file.name());
+      continue;
+    }
+
+    credentials.push_back(creds);
+  }
+
+  return true;
+}
+
+WiFiCredentials::WiFiCredentials(std::uint8_t id, const wifi_ap_record_t* record) {
+  _id = id;
+
+  static_assert(sizeof(_ssid) == sizeof(record->ssid), "WiFiCredentials::_ssid and wifi_ap_record_t::ssid must be the same size");
+  memset(_password, 0, sizeof(_password));
+  memcpy(_ssid, record->ssid, sizeof(_ssid));
+}
+
+WiFiCredentials::WiFiCredentials(std::uint8_t id, const String& ssid, const String& password) {
+  _id = id;
+
+  auto ssidLength = ssid.length();
+  if (ssid.length() > sizeof(_ssid)) {
+    ESP_LOGW(TAG, "SSID is too long, truncating");
+    ssidLength = sizeof(_ssid) - 1;
+  }
```

Comment:
Does that comparison work? It strikes me as ...odd. To me it looks like you're comparing a string length against a pointer size.

Reply:
sizeof(_ssid) gets the size of the array, but i did get the size of the array wrong, appended -1